.github/workflows/ci.yml

```diff
@@ -39,4 +37,67 @@ jobs:
     - name: Lint Python Code
       run: |
             pip install ruff
-            ruff check
\ No newline at end of file
+            ruff check
+
+  demo:
+    name: Demo Job
+    runs-on: ${{ matrix.os }}
+    strategy: 
+      matrix:
+        os: [ubuntu-latest]
```

Comment:
Can we add macos-latest alongside ubuntu-latest for the operating system?

Reply:
Yes I'll get that added!

Reply:
I'm having a bit of trouble trying to get the `macos-latest` version running. Can you take a look at [this check](https://github.com/RustomAlexios/IMPALIB/actions/runs/9895263723/job/27334739305?pr=14) with the error under `Demo Job (macos-latest)` -> `Run CPP Demos` -> `Line 406`

Perhaps something in the `CMakeLists.txt` that needs to be updated? Its difficult for me to debug as I'm not on a mac currently

Reply:
Can you try adding this in CMakeLists.txt:

add_compile_options(-std=c++11) #added for Mac recognition

or 

set(CMAKE_CXX_STANDARD 11)

Do you have the latest CMakeLists.txt from main?

Reply:
This didn't see to do the trick. I'm still looking into this currently

Reply:
What the error that you are getting?

Reply:
Everything is working for me okay locally, on WSL. However the CI doesn't seem to work for the Header Only CPP jobs on `mac-os`. The error can be seen in this job -> https://github.com/RustomAlexios/IMPALIB/actions/runs/9895263723/job/27334739305?pr=14

Reply:
![image](https://github.com/user-attachments/assets/3c2339a3-acf9-476a-a622-674ca0dda735)

Reply:
I realized that the issue on my Mac was resolved by following these steps:

1) Activate environment X.
2) Run python3 -m pip install . -v while environment X is active.
3) Move to the directory examples/KcMwm.
4) Ensure that environment X is still activated before running cmake -B build ..

This gave no error.

However, if you activate another environment instead of environment X (after deleting the generated build file), I encountered the same error you had above. (Note that the build file has to be deleted so this error can be observed). 

Can we create and activate a Python virtual environment in Github workflow where we can install all dependencies? (maybe this could resolve the issue) (https://stackoverflow.com/questions/74668349/how-to-activate-a-virtualenv-in-a-github-action)

Although find_package(impalib REQUIRED) is also in the CMakeLists.txt inside examples/KcMwm, we still get an error.

Not sure if we also need to delete build files before running the remaining checks after we do changes. What do you think?

Reply:
@RustomAlexios 

Okay, I think I've solved the issues with getting both `mac-os` and `ubuntu` working. I went ahead and added another [github action repository](https://github.com/marketplace/actions/setup-miniconda) and added calls to activate the conda environment before running each of the demo jobs. Take a look and let me know what you think

Also, sorry if your inbox has been flooded with notifications from all my pushes. I set the PR back to "Draft" mode, however I don't know if on github that silences the notifications for you